### DISCUSSION TITLE
Implementing chiral EFTs in FeynCalc

### DISCUSSION THREAD

**Discussion by AlexLong152** [FeynCalc/feyncalc#244] [Q&A]
Hello, I am interested in implementing diagrams such as the one below which shows pion scattering off of two nucleons. 
Are there any examples that implement something like this? Any recommendations on how to do this, and other similar diagrams?

![diagram-20240129](https://github.com/FeynCalc/feyncalc/assets/49763479/33327eef-e78d-46fb-9d03-80395272de35)

**Comment by vsht** [comment]
Hi,

well, you certainly could use QGRAF to implement your model and generate your diagrams 

https://feyncalc.github.io/FeynHelpersBookDev/Extra/QGRAFUsageExamples.html

Whether the output can be conveniently calculated using FeynCalc is of course a totally different question.
Apart from that, I have no recommendations or pointers to similar projects.

**Reply by AlexLong152** [reply]
Thank you for pointing out QGRAF to me, I didn't know that existed. 

I am attempting to calculate Pion-Pion scattering in chiral perturbation theory. The Feynman rules for this theory can be found in appendix A of this pdf:  https://arxiv.org/abs/hep-ph/9501384v1.

Do you have any recommendations on how to proceed with this? The learning curve seems rather steep.

**Comment by vsht** [comment]
>> 
>> Do you have any recommendations on how to proceed with this? The 
>> learning curve seems rather steep.
>> 

QGRAF is pretty much the only way to get the generation of such 
nonstandard diagrams automated, so learning how to use is a valuable
investment that will pay off in many ways. BTW, it's also the standard
tool for almost all multiloop calculations.

The model itself should be pretty easy to implement since you don't need
to specify any Feynman rules at that point.

Inserting Feynman rules and visualizing the diagrams is something that
requires custom codes built around QGRAF. FeynHelpers offers one way
of doing that, but there are other options like Alibrary, tapir,
Feamngen.jl, HepLib etc.

There's no shortcut to get your diagrams generated without understanding
how the underlying machinery works. So if it's important to get those
calculations automated, you'll have to spend some time dealing with the 
codes.

If it's just a gimmick that has no real importance, I'd rather write 
down the amplitudes by hand, which should be much faster.




Am 26.02.24 um 19:10 schrieb Alex Long:
